support/gitignore

```diff
@@ -39,3 +39,27 @@ CPPLINT.cfg
 .deps/
 .libs/
 *-stamp
+
+.clang-format
+.clang-tidy
+.gitlint
+.pre-commit-config.yaml
+.reviewboardrc
+*/**/Makefile.in
+CPPLINT.cfg
+Makefile.in
+aclocal.m4
+ar-lib
+autom4te.cache/
+build/
+centos7/
+compile
+config.guess
+config.sub
+configure
+*~
```

Comment:
This `*~` will catch many files, are there many such files which can't be listed individually?

Reply:
I changed my mind, I think that does not have to be there. With that I want catch tmp files from my editor. But maybe that's to much. :sweat_smile: